modules/seqexec/server/src/main/scala/seqexec/server/SeqTranslate.scala

```diff
@@ -92,45 +81,52 @@ trait SeqTranslate[F[_]] extends ObserveActions {
 
 object SeqTranslate {
   private class SeqTranslateImpl[F[_]: Sync: Logger](site: Site,
-                                                     systems: Systems[F],
+                                                     systemss: Systems[F],
                                                      gmosNsCmd: Ref[F, Option[NSObserveCommand]]
                                                     ) extends SeqTranslate[F] {
 
+    private val overriddenSystems = new OverriddenSystems[F](systemss)
+
     private def step(obsId: Observation.Id, i: StepId, config: CleanConfig, nextToRun: StepId,
                      datasets: Map[Int, ExecutedDataset], isNightSeq: Boolean)(
                        implicit cio: Concurrent[F],
                                 tio: Timer[F]
-                     ): F[SequenceGen.StepGen[F]] = {
+                     ): F[StepGen[F]] = {
       def buildStep(
         dataId: DataId,
-        inst: InstrumentSystem[F],
-        sys: List[System[F]],
-        headers: HeaderExtraData => List[Header[F]],
+        instRes: Resource,
+        insSpec: InstrumentSpecifics,
+        instf: SystemOverrides => InstrumentSystem[F],
+        otherSysf: Map[Resource, SystemOverrides => System[F]],
```

Comment:
It could be, but that would introduce some problems. The first one is that all the systems in `otherSysf` are treated generically, but we need to access the special behavior of `InstrumentSystem`. Also, the `InstrumentSystem` is mandatory, and that will not be reflected if we pass it in `otherSysf`.